client/db/src/offchain.rs

```diff
@@ -100,7 +105,7 @@ impl sp_core::offchain::OffchainStorage for LocalStorage {
 			is_set = val.as_ref().map(|x| &**x) == old_value;
 
 			if is_set {
-				self.set(prefix, item_key, new_value)
```

Comment:
Is it guaranteed that the error here would quit the app? Cause we fail to update `self.locks` in case of error, so the lock will keep hanging there. I'd rather handle that gracefuly (i.e. release the lock and then return an error).

Reply:
Ok I note that. I need to check again that changing this trait was necessary for the clean exit because I'm not 100% sure. I've put too much time between the moment I started this PR and the moment I continued.

Reply:
It wasn't needed at all. I made a mistake. I reverted all of that